rudder-core/src/main/scala/com/normation/rudder/domain/policies/Tags.scala

```diff
@@ -44,11 +44,15 @@ import com.normation.utils.Control._
  * Tags that apply on Rules and Directives
  * We do not warranty unicity of tags name, only of tagsname + tagvalue
  */
-final case class TagName( name : String )
 
-final case class TagValue( value : String )
+object Tag {
+  final implicit class TagName ( val value : String )
```

Comment:
I think we really need them to be implicit *case* class, else the equals/hashcode won't be what we want at all